File: Code/3DMANAGER-APP/3dmanager-app.client/src/models/user/UserDetailObject.ts
import type { FileResponse } from "../file/fileResponse";

export interface UserDetailObject {
    userId: number;
    userName: string;
    userRole: string;
    userCreateDate: Date;
    userEmail: string;
    userPrintHours: string;
    userPrintedPrints: number;
    userTotalHours: string;
    userTotalPrints: number;
    userImageData?: FileResponse;
}

File: Code/3DMANAGER-APP/3DMANAGER-APP.BLL/Models/User/UserDetailObject.cs
using _3DMANAGER_APP.BLL.Models.File;

namespace _3DMANAGER_APP.BLL.Models.User
{
    public class UserDetailObject
    {
        public int userId { get; set; }
        public string userName { get; set; }
        public string userRole { get; set; }
        public string userEmail { get; set; }
        public DateTime UserCreateDate { get; set; }
        public int UserTotalPrints { get; set; }
        public string UserTotalHours { get; set; }
        public int UserPrintedPrints { get; set; }
        public string UserPrintHours { get; set; }
        public FileResponse? UserImageData { get; set; }
    }
}
